### ISSUE TITLE
Add Mono Game Unstripping

### ISSUE THREAD

**Issue by HerpDerpinstine** [LavaGang/MelonLoader#100]
Currently certain Mono based Games come with stripped DLLs in their Managed folder.
This can cause initialization failures with MelonLoader.

An automatic built-in preliminary pass of the Managed folder to unstrip these when a Mono game is detected would be ideal.
Similar to how there is already a preliminary pass to unstrip DLLs on Il2Cpp but for Mono Games.
Most likely needing to destroy the Mono Domain, replacing the files, then recreating the Mono Domain.

Temporary workaround is to manually replace the stripped files with unstripped ones.

**Comment by sebastientromp** [comment]
Would it be possible to have a hook when MelonLoader starts, but before it loads anything? In my specific case, always copying the DLL from the MelonLoader/Managed folder to the game's Managed folder would be an acceptable workaround

**Comment by slxdy** [comment]
Considering that the managed part of MelonLoader doesn't have many System calls anymore, it's worth retesting on the failing games. Otherwise, we could always move some calls to the bootstrap

**Comment by HerpDerpinstine** [comment]
> Considering that the managed part of MelonLoader doesn't have many System calls anymore, it's worth retesting on the failing games. Otherwise, we could always move some calls to the bootstrap

This is also for Mods as well.

**Comment by slxdy** [comment]
Fair enough, but how do you wanna go about doing this? We'd have to ship System assemblies of all .net versions.
Does Unity mono ship builds through github or some other open API? We could technically download the right version whenever needed
